protocol/transport.go

```diff
@@ -5,6 +5,20 @@ import (
 	"io"
 )
 
+// TransactionState is used as a return with every message read for commit and rollback implementation
+type TransactionState int
+
+const (
+	// TransactionEnded states that the current transaction has finished and has to commit
+	TransactionEnded TransactionState = 1 + iota
+	// TransactionFailed states that the current transaction has failed and has to roll-back
+	TransactionFailed
+	// InTransaction states that transaction is active and operations should not commit
+	InTransaction
+	// NotInTransaction states that transaction is not active and operations should auto-commit
+	NotInTransaction
```

Comment:
What do you think about reordering these values according to the natural order? Not in, in, ended, failed?